File: LatinReader/app/src/main/java/com/ericmschmidt/latinreader/activities/ErrorActivity.java
package com.ericmschmidt.latinreader.activities;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.ericmschmidt.classicsreader.R;

public class ErrorActivity extends AppCompatActivity {

    public final static String ERROR_KEY = "com.ericmschmidt.latinreader.ERROR";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_error);
        TextView errorMessageText = (TextView) findViewById(R.id.error_activity_content);
        String errorMessage = getString(R.string.default_error_message);;

        Bundle extras = getIntent().getExtras();

        if ((savedInstanceState != null) &&
                savedInstanceState.containsKey(ERROR_KEY)) {
            errorMessage = savedInstanceState.getString(ERROR_KEY);

        } else if (extras != null) {
            errorMessage = extras.getString(ERROR_KEY);

        }

        errorMessageText.setText(errorMessage);
    }
}

File: LatinReader/views/src/main/java/com/ericmschmidt/classicsreader/activities/ErrorActivity.kt
package com.ericmschmidt.classicsreader.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.ericmschmidt.classicsreader.R
import com.ericmschmidt.classicsreader.views.databinding.ActivityErrorBinding

/**
 * Error message activity for this app.
 * @author Eric Schmidt
 * @author <a href="https://telpirion.com">...</a>
 * @version 1.5
 * @since 1.1
 */
class ErrorActivity : AppCompatActivity() {

    private lateinit var binding: ActivityErrorBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityErrorBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val errorMessage = savedInstanceState?.getString(ERROR_KEY)
            ?: intent.extras?.getString(ERROR_KEY)
            ?: getString(R.string.greekreader_default_error_message)

        binding.errorActivityContent.text = errorMessage
    }

    companion object {
        const val ERROR_KEY = "com.ericmschmidt.latinreader.ERROR"
    }
}
